repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphMapper.java

```diff
@@ -2809,12 +2810,21 @@ public void deleteClassification(String entityGuid, String classificationName) t
                             .anyMatch(x -> classificationHasPendingTask(x, classificationVertexId, entityGuid));
 
                     if (pendingTaskExists) {
-                        LOG.error("Another tag propagation is in queue for classification: {} and entity: {}. Please try again", classificationVertexId, entityGuid);
-                        throw new AtlasBaseException(AtlasErrorCode.DELETE_TAG_PROPAGATION_NOT_ALLOWED, classificationVertexId, entityGuid);
+                        List<AtlasTask> entityClassificationPendingTasks = entityPendingTasks.stream()
+                                .filter(t -> t.getParameters().get("entityGuid").equals(entityGuid)
+                                        && t.getParameters().get("classificationVertexId").equals(classificationVertexId))
```

Comment:
Certain types of tasks might not have entityGuid, classificationVertexId defined in parameters, which will result in an NPE.
Add check for task type as well - CLASSIFICATION_PROPAGATION_ADD

Reply:
When I add check for task type CLASSIFICATION_PROPAGATION_ADD, NPE situation is fixed on its own or do I handle it separately? I mean, do all CLASSIFICATION_PROPAGATION_ADD types have entityGuid and classificationVertexId?